dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/FourEyeApproverActionlet.java

```diff
@@ -0,0 +1,166 @@
+
+    private static final String ACTIONLET_NAME = "'4 Eye' Approval";
+    private static final String HOW_TO =
+            "This actionlet implements the '4 Eyes' principle for verifying new content. It takes a comma-separated list of user IDs, email addresses, or role keys "
+                    + "(i.e., the users assigned to those roles) which will be in charge of reviewing the content, and a minimum number of approvers. "
+                    + "If the number of approvers is greater or equal than the specified value, the next sub-action will be executed.";
+    private static final String USER_ID_DELIMITER = ",";
+    private static final String PARAM_CONTENT_APPROVERS = "approvers";
+    private static final String PARAM_MINIMUM_CONTENT_APPROVERS = "minimumApprovers";
+    private static final String PARAM_EMAIL_SUBJECT = "emailSubject";
+    private static final String PARAM_EMAIL_BODY = "emailBody";
+    private static final String PARAM_IS_HTML = "isHtml";
+
+    private static final int DEFAULT_MINIMUM_CONTENT_APPROVERS = 2;
+
+    private boolean shouldStop = false;
+
+    private static ArrayList<WorkflowActionletParameter> ACTIONLET_PARAMETERS = null;
+
+    @Override
+    public synchronized List<WorkflowActionletParameter> getParameters() {
+        if (null == ACTIONLET_PARAMETERS) {
+            ACTIONLET_PARAMETERS = new ArrayList<>();
+            ACTIONLET_PARAMETERS
+                    .add(new MultiUserReferenceParameter(PARAM_CONTENT_APPROVERS,
+                            "User IDs, Emails, or Role Keys", null,
+                            true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_MINIMUM_CONTENT_APPROVERS,
+                            "Number of Approvers",
+                            String.valueOf(DEFAULT_MINIMUM_CONTENT_APPROVERS), true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_EMAIL_SUBJECT,
+                            "Email Subject",
+                            "'4 Eye' Approval Required", true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_EMAIL_BODY, "Email Message",
+                            null,
+                            false));
+        }
+        return ACTIONLET_PARAMETERS;
+    }
+
+    @Override
+    public String getName() {
+        return ACTIONLET_NAME;
+    }
+
+    @Override
+    public String getHowTo() {
+        return HOW_TO;
+    }
+
+    @Override
+    public boolean stopProcessing() {
+        return this.shouldStop;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (null != obj && obj instanceof WorkFlowActionlet) {
+            return getClass().equals(obj.getClass());
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return (shouldStop ? 1 : 0);
+    }
+
+    @Override
+    public void executeAction(final WorkflowProcessor processor,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)